File: .idea/app/src/main/java/com/chevstrap/rbx/Models/APIs/IPInfoResponse.java
package com.chevstrap.rbx.Models.APIs;

import org.json.JSONObject;

public class IPInfoResponse {
    private final String city;
    private final String region;
    private final String country;

    public IPInfoResponse(String city, String region, String country) {
        this.city = city;
        this.region = region;
        this.country = country;
    }

    public String getCity() { return city; }
    public String getRegion() { return region; }
    public String getCountry() { return country; }

    public static IPInfoResponse parse(String json) {
        try {
            JSONObject jsonObject = new JSONObject(json);
            String city = jsonObject.optString("city", "");
            String region = jsonObject.optString("region", "");
            String country = jsonObject.optString("country", "");
            return new IPInfoResponse(city, region, country);
        } catch (Exception e) {
            return null;
        }
    }
}

File: app/src/main/java/com/chevstrap/rbx/models/apis/IPInfoResponse.kt
package com.chevstrap.rbx.models.apis

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.json.JSONObject

@Serializable
data class IPInfoResponse(
    @SerialName("city")
    val city: String? = null,

    @SerialName("country")
    val country: String? = null,

    @SerialName("region")
    val region: String? = null
)